purpleair-aqi.js

```diff
@@ -41,15 +42,93 @@ const SENSOR_ID = args.widgetParameter || "69223";
  * @property {string} lon
  */
 
+/**
+ * @typedef {object} LatLon
+ * @property {number} latitude
+ * @property {number} longitude
+ */
+
+/**
+ * Get the closest PurpleAir sensorId to the given location
+ *
+ * @param {LatLon} location
+ * @returns {Promise<number>}
+ */
+async function getSensorId({ latitude, longitude }) {
+  if (SENSOR_ID) return SENSOR_ID;
+
+  const BOUND_OFFSET = 0.05;
```

Comment:
This determines the size of the box in which to look for sensors. The larger the `BOUND_OFFSET`, the larger the box. This _might_ be too large in a built-up area, or too small in a more rural area, depending on the sensor density.

It might be worth experimenting to determine the best bounding size.